File: stock_master_backend/src/main/java/com/company/stock_master/master/stock/Stock.java
package com.company.stock_master.master.stock;

import com.company.stock_master.master.product.Product;
import com.company.stock_master.master.warehouse.entity.Location;
import com.company.stock_master.master.warehouse.entity.Warehouse;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "stock")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Stock {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "location_id", nullable = false)
    private Location location;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "warehouse_id", nullable = false)
    private Warehouse warehouse;

    private long onHand;
    private long reserved;
    private long freeToUse;
}

File: frontend/src/types/Stock.ts
import type { Product } from './Product'
import type { Warehouse } from './Master/Warehouse'
import type { Location } from './Master/Location'

export interface Stock {
  id: number
  productId: number
  warehouseId: number
  locationId: number
  onHand: number
  reserved: number
  freeToUse: number
  product?: Product
  warehouse?: Warehouse
  location?: Location
}
